exercises/rational-numbers/src/test/java/RationalTest.java

```diff
@@ -0,0 +1,293 @@
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class RationalTest {
+
+    // Helper methods
+
+    private static final double DOUBLE_EQUALITY_TOLERANCE = 1e-15;
+
+    private void assertDoublesEqual(double x, double y) {
+        assertEquals(x, y, DOUBLE_EQUALITY_TOLERANCE);
+    }
+
+    private void assertRationalsEqual(Rational r1, Rational r2) {
+        assertEquals(r1.getNumerator(), r2.getNumerator());
+        assertEquals(r1.getDenominator(), r2.getDenominator());
```

Comment:
Will this give a good error message if the test fails? That is will it clearly tell the user that it's the denominator or the numerator that's wrong or will it just say something like `Expected 5 but got 2`?

Reply:
@FridaTveit It won't, no. It looks like this:
```
RationalTest > testAddTwoPositiveRationalNumbers FAILED
    java.lang.AssertionError: expected:<9> but was:<6>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at RationalTest.assertRationalsEqual(RationalTest.java:18)
        at RationalTest.testAddTwoPositiveRationalNumbers(RationalTest.java:27)
```

I cribbed the `assertRationalsEqual` helper method from the `assertComplexNumbersEqual` method in `ComplexNumberTest.java` in `complex-numbers`. Would you like it to be more specific? I suppose we could implement `equals` and maybe `toString` on `Rational`, or require the exercise-taker to do it. That way, we could get better error messages.

Reply:
You could implement `equals`, that would help. In which case I would add some starter implementation with `equals` already implemented since that's not really part of the exercise. Or you could just add a message to your equals check saying something like `Incorrect numerator`. I think either would be fine 🙂 

You're right, it is the same in complex-numbers, thanks for pointing it out! I'll open an issue to improve the error message for complex-numbers 🙂